module/component/component.go

```diff
@@ -146,18 +147,44 @@ func RunComponent(ctx context.Context, componentFactory ComponentFactory, handle
 type ReadyFunc func()
 
 // ComponentWorker represents a worker routine of a component.
-// It takes a SignalerContext which can be used to throw any irrecoverable errors it encouters,
+// It takes a SignalerContext which can be used to throw any irrecoverable errors it encounters,
 // as well as a ReadyFunc which must be called to signal that it is ready. The ComponentManager
-// waits until all workers have signaled that they are ready before closing its own Ready channel.
+// waits until all workers have signaled that they are ready before closing its Component's
+// Ready channel.
 type ComponentWorker func(ctx irrecoverable.SignalerContext, ready ReadyFunc)
 
+// ComponentTask represents a one-off task.
+// It takes a SignalerContext which can be used to throw any irrecoverable errors it encounters.
+type ComponentTask func(ctx irrecoverable.SignalerContext)
+
 // ComponentManagerBuilder provides a mechanism for building a ComponentManager
 type ComponentManagerBuilder interface {
 	// AddWorker adds a worker routine for the ComponentManager
 	AddWorker(ComponentWorker) ComponentManagerBuilder
 
 	// Build builds and returns a new ComponentManager instance
-	Build() *ComponentManager
+	Build() ComponentManager
+}
+
+// ComponentManager manages the worker routines and tasks of a Component.
+type ComponentManager interface {
+	Component() Component
```

Comment:
I originally had it the way you described, but later intentionally changed it to separate the two. The intention is precisely that the engine only embeds the `Component`, and it stores the `ComponentManager` as a separate private field.

Otherwise, all of the `ComponentManager`'s methods (`Run`, `RunAsync`) will be directly accessible via the engine, which is unnecessary / undesirable. Only the engine itself should have access to the `ComponentManager`.